### PR TITLE
refactor: update `stats/base/dmax` native addon from C++ to C

### PR DESCRIPTION
**Pull request by aayush0325**
Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   Refactors the native addon of `stats/base/dmax` to follow latest conventions.
-   I browsed through some old PRs ( following #1807 here ) to get an idea of what should be done here. 
-   **So far, the only change made here is migration from C++ to C addons**
-   However for refrence say #1807 has some other changes related to tests, benchmarks and documentation as well. Please let me know if we need to do the same here

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dmax/manifest.json [hunk 1]
```diff
@@ -24,6 +26,28 @@
...
+        "@stdlib/math/base/assert/is-positive-zero",
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv",
+        "@stdlib/napi/argv-int64",
+        "@stdlib/napi/argv-strided-float64array"
+      ]
+    },
+    {
+      "task": "examples",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dmax/manifest.json:68]: You also need a configuration for benchmarks.

File: lib/node_modules/@stdlib/stats/base/dmax/src/addon.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, stride, argv, 1 );
+
+	napi_value v;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dmax/src/addon.c]: Use `STDLIB_NAPI_CREATE_DOUBLE`.

Comment (kgryte) [reply lib/node_modules/@stdlib/stats/base/dmax/src/addon.c]: You should be able to search `blas/ext/base/*` for an example usage.

**Review by kgryte — CHANGES_REQUESTED** [review]
This is looking good. Just two minor comments.

**Comment by kgryte** [comment]
Let's hold off on additional changes for future PRs.

**Comment by aayush0325** [comment]
Noted, pushing the requested changes soon!

**Comment by aayush0325** [comment]
is this good to go @kgryte?

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aayush0325!

**Pull request merged**